jena-arq/src/main/java/org/apache/jena/sparql/util/FmtUtils.java

```diff
@@ -477,27 +477,56 @@ private static boolean checkValidLocalname(String localname) {
 
         for ( int idx = 0 ; idx < localname.length() ; idx++ ) {
             char ch = localname.charAt(idx);
-            if ( !validPNameChar(ch) )
-                return false;
+            if ( idx == 0 ) {
+                if ( !validFirstPNameChar(ch) )
+                    return false;
+            } else if ( idx == localname.length() - 1 ) {
+                if ( !validEndPNameChar(ch) )
+                    return false;
+            } else {
+                if ( !validMidPNameChar(ch) )
+                    return false;
+            }
         }
+        return true;
+    }
 
-        // Test start and end - at least one character in the name.
+    // first char as per options in first bracketed production in 169: https://www.w3.org/TR/sparql11-query/#rPN_LOCAL
+    private static boolean validFirstPNameChar(char ch) {
+        return validPNameCharU(ch) || ( ch >= '0' && ch <= '9' ) || ( ch == '%' ) ;
+    }
 
-        if ( localname.endsWith(".") )
-            return false;
-        if ( localname.startsWith(".") )
-            return false;
+    // neither first char nor last in middle repeated production in 169: https://www.w3.org/TR/sparql11-query/#rPN_LOCAL
+    private static boolean validMidPNameChar(char ch) {
+        return validPNameChar(ch) || ( ch == '.' ) || ( ch == ':' ) || ( ch == '%' ) ;
+    }
 
-        return true;
+    // last char, but not first, in last production in 169: https://www.w3.org/TR/sparql11-query/#rPN_LOCAL
+    private static boolean validEndPNameChar(char ch) {
+        return validPNameChar(ch) || ( ch == ':' ) || ( ch == '%' ) ;
+    }
+
+    // production 164: https://www.w3.org/TR/sparql11-query/#rPN_CHARS_BASE
```

Comment:
There are functions to test for the token rules in `RiotChars`.

This one is `RiotChars.isPNCharsBase`.